x-pack/legacy/plugins/ml/server/routes/results_service.ts

```diff
@@ -0,0 +1,220 @@
+  return rs.getCategoryDefinition(payload.jobId, payload.categoryId);
+}
+
+function getCategoryExamples(context: RequestHandlerContext, payload: any) {
+  const rs = resultsServiceProvider(context);
+  const { jobId, categoryIds, maxExamples } = payload;
+  return rs.getCategoryExamples(jobId, categoryIds, maxExamples);
+}
+
+function getMaxAnomalyScore(context: RequestHandlerContext, payload: any) {
+  const rs = resultsServiceProvider(context);
+  const { jobIds, earliestMs, latestMs } = payload;
+  return rs.getMaxAnomalyScore(jobIds, earliestMs, latestMs);
+}
+
+function getPartitionFieldsValues(context: RequestHandlerContext, payload: any) {
+  const rs = resultsServiceProvider(context);
+  const { jobId, searchTerm, criteriaFields, earliestMs, latestMs } = payload;
+  return rs.getPartitionFieldsValues(jobId, searchTerm, criteriaFields, earliestMs, latestMs);
+}
+
+/**
+ * Routes for results service
+ */
+export function resultsServiceRoutes({ xpackMainPlugin, router }: RouteInitialization) {
+  /**
+   * @apiGroup ResultsService
+   *
+   * @api {post} /api/ml/results/anomalies_table_data Prepare anomalies records for table display
+   * @apiName GetAnomaliesTableData
+   * @apiDescription Retrieves anomaly records for an anomaly detection job and formats them for anomalies table display
+   */
+  router.post(
+    {
+      path: '/api/ml/results/anomalies_table_data',
+      validate: {
+        body: schema.object({ ...anomaliesTableDataSchema }),
+      },
+    },
+    licensePreRoutingFactory(xpackMainPlugin, async (context, request, response) => {
+      try {
+        const resp = await getAnomaliesTableData(context, request.body);
+
+        return response.ok({
+          body: resp,
+        });
+      } catch (e) {
+        return response.customError(wrapError(e));
+      }
+    })
+  );
+
+  /**
+   * @apiGroup ResultsService
+   *
+   * @api {post} /api/ml/results/category_definition Returns category definition
+   * @apiName GetCategoryDefinition
+   * @apiDescription Returns the definition of the category with the specified ID and job ID
+   */
+  router.post(
+    {
+      path: '/api/ml/results/category_definition',
+      validate: {
+        body: schema.object({
+          jobId: schema.maybe(schema.string()),
+          categoryId: schema.string(),
+        }),
+      },
+    },
+    licensePreRoutingFactory(xpackMainPlugin, async (context, request, response) => {
+      try {
+        const resp = await getCategoryDefinition(context, request.body);
+
+        return response.ok({
+          body: resp,
+        });
+      } catch (e) {
+        return response.customError(wrapError(e));
+      }
+    })
+  );
+
+  /**
+   * @apiGroup ResultsService
+   *
+   * @api {post} /api/ml/results/max_anomaly_score Returns the maximum anomaly_score
+   * @apiName GetMaxAnomalyScore
+   * @apiDescription Returns the maximum anomaly_score for result_type:bucket over jobIds for the interval passed in
```

Comment:
Edit to `Returns the maximum anomaly score of the bucket results for the request job ID(s) and time range.`

Reply:
Updated 0574d51d9d2b613b8d0c7ecf60385eddb433de33